acl-batch/main.tf

```diff
@@ -0,0 +1,14 @@
+locals {
+  offset = var.index * var.acl_count
+}
+
+resource "kafka_acl" "kafka_acl" {
+  count                        = min(var.max_acl, var.max_acl - local.offset)
```

Comment:
This should prolly make sure that max_acl - offset is positive or 0...

Reply:
So turns out this logic was incorrect and should have been:

`min(var.acl_count, var.max_acl - local.offset)`

for less confusion (ie so the submodule matches the parent) , I have renamed `acl_count` to `acls_per_batch`, so it's now:

`min(var.acls_per_batch, var.max_acl - local.offset)`

offset is calculated as: `offset = var.index * var.acls_per_batch`
`acls_per_batch` already has validation on it (0 < var.acls_per_batch <= 5000).
I've just added validation to `index` (index >= 0) so offset can't go negative (although I guess could become very positive).
I've also added validation to `max_acl` so it can't go negative either

Last way it could get negative is if `offset > max_acl` == `index * acls_per_batch > max_acl`
Highest value for `index` = `number_of_batches - 1`.
`number_of_batches = ceil(var.max_acl / var.acls_per_batch)`

Highest value for `index` = `ceil(var.max_acl / var.acls_per_batch) - 1`.

`index * acls_per_batch > max_acl`
`(ceil(var.max_acl / var.acls_per_batch) - 1) * acls_per_batch > max_acl`
`(var.max_acl / var.acls_per_batch) * acls_per_batch > max_acl` # Remove ceil & -1
`var.max_acl > max_acl` # Simplify

Therefore, should be impossible unless max_acl is ever > max_acl (or code is changed?...).